apis/nucleus/src/viz.js

```diff
@@ -149,6 +150,46 @@ export default function viz({ model, halo, initialError, onDestroy = async () =>
       const propertyTree = await conversionConvertTo({ halo, model, cellRef, newType });
       return propertyTree;
     },
+    /**
+     * Toggles the chart data view
+     *
+     * @experimental
+     * @since 4.7.0
+     * @param {boolean=} showDataView - If included, turns the toggle into a one way-only operation. If true it will only toggle to the view data table.
+     */
+    async toggleDataView(showDataView) {
+      let newModel;
+      if (!viewDataObjectId && showDataView !== false) {
+        const oldProperties = await model.getEffectiveProperties();
+        const propertyTree = await conversionConvertTo({
+          halo,
+          model,
+          cellRef,
+          newType: halo.config.context.dataViewType,
+          properties: oldProperties,
+        });
+        const newProperties = { ...propertyTree.qProperty, totals: { show: false }, usePagination: true };
+        newModel = await halo.app.createSessionObject(newProperties);
+        viewDataObjectId = newModel.id;
+      } else if (viewDataObjectId && showDataView !== true) {
+        newModel = model;
+        await halo.app.destroySessionObject(viewDataObjectId);
+        viewDataObjectId = undefined;
+      }
+      if (newModel) {
+        await unmountCell();
+        [unmountCell, cellRef] = glueCell({
```

Comment:
This actually creates a cell every time. Would there be any downside to this?

Reply:
Compared to conversion you create a new model, so it won't automagically update the current Cell. 
But what you could do is replace what model the Cell uses. Either you'd have to restructure how the React stuff is setup OR possibly add a setModel function to the importativeHandle (cellRef) and replace it that way

Reply:
So went with the second option of adding a setModel function. Not sure how we would restructure the React setup. The state will need to be handled inside the React parts if we want to keep it as state. Otherwise I think we still have to setup the portal every single time we toggle